src/Init/Data/String/Basic.lean

```diff
@@ -89,13 +123,38 @@ def utf8SetAux (c' : Char) : List Char → Pos → Pos → List Char
   | c::cs, i, p =>
     if i = p then (c'::cs) else c::(utf8SetAux c' cs (i + c) p)
 
+/--
+Replaces the character at a specified position in a string with a new character.
+
+If the position is invalid, returns the string unchanged.
```

Comment:
```suggestion
/--
Replaces the character at a specified position in a string with a new character. If the position is invalid, the string is returned unchanged.

If both the replacement character and the replaced character are ASCII characters and the string is not shared, destructive updates are used.
```

Reply:
I think it makes sense to move this here, because it's still describing the behavior. With this structure, the first paragraph is a functional spec, and the second describes relevant nonfunctional properties.